### ISSUE TITLE
Futures `Hello World` example (same as in `Hello World!` section)

### ISSUE THREAD

**Issue by goriunov** [tokio-rs/doc-push#52]
I have went trough the guide and noticed that we have `Hello World!` example which uses tokio directly with the tasks. Would be good to see exactly the same example but implemented with `futures (stream)` directly  so people will be able to compare it and see what tokio does on background. I also went through the Futures section and it uses a little different example which is feels like completely separate thing and just confuses me more like why would.

I believe it would be good to see different ways of implementing the same thing to understand how everything is related in this case (Futures (Stream), Tokio, Tasks)

**Comment by goriunov** [comment]
Would be actually good to have few examples which uses different approaches (more low and high levell)

**Comment by carllerche** [comment]
The example on this page? https://tokio.rs/docs/getting-started/futures/

Also, are you suggesting that there should be more examples that use combinators and then repeat the example by implementing `Future` directly?

**Comment by goriunov** [comment]
> Also, are you suggesting that there should be more examples that use combinators and then repeat the example by implementing Future directly?

Yes at least few examples (simple, intermediate) which would be implemented with `combinators` and re implemented exactly the same in `futures`. This will help to understand more how the system works (especially for beginners as it is very hard to find a reason for Futures and Combinators and how to use each of them and in which situation and how really Combinators work with Futures with out having exactly the same app implemented in different approaches, at least for me). I am still a beginner in Tokio so i may say some things wrong.

**Comment by carllerche** [comment]
Thoughts @rylev ?

**Comment by rylev** [comment]
I agree that this would be useful though with the reworking of the Getting Started section in https://github.com/tokio-rs/website/pull/224 I don't think that Getting Started is the right place to do this. I think we should find a place to do this in the "Futures, Streams and Sinks" guide.

**Comment by carllerche** [comment]
We probably want to revisit the getting started examples in "Futures, Streams, and Sinks" to do the compare / contrast.
